lib/libc/glibc/sysdeps/generic/sysdep.h

```diff
@@ -56,6 +56,12 @@
 # define cfi_personality(enc, exp)	.cfi_personality enc, exp
 # define cfi_lsda(enc, exp)		.cfi_lsda enc, exp
 
+# ifndef __clang__
```

Comment:
I don't see this in upstream glibc. Should a patch be submitted?

Reply:
Maybe? Idk if we've ever sent a patch to glibc before.

Reply:
Done.

Reply:
Ok, something is weird here:

```console
❯ clang --version
Ubuntu clang version 16.0.6 (15)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
❯ clang -target riscv test.s -c
❯ clang -target riscv64 test.s -c
test.s:5:1: error: unknown directive
.cfi_label .Ldummy
^
```

Reply:
Clang doesn't support the directive which is why we have this.

Reply:
https://github.com/llvm/llvm-project/issues/97222

Reply:
> Clang doesn't support the directive which is why we have this.

It's definitely an oversight. The `MCStreamer` code in LLVM has an API for emitting CFI labels, it's just that the `MCParser` hasn't been taught to parse them. Honestly a bit a surprised no one has run into this before as it affects ARC, C-SKY, and LoongArch too.

Reply:
We should be able to drop this workaround in LLVM 19: https://github.com/llvm/llvm-project/pull/97922

Reply:
Sweet, amazing getting that done. Would that mean this PR will have to wait for the next LLVM version bump in Zig?